frontend/discourse/admin/templates/admin-config/login/discourse-id.gjs

```diff
@@ -0,0 +1,204 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import DButton from "discourse/components/d-button";
+import DToggleSwitch from "discourse/components/d-toggle-switch";
+import icon from "discourse/helpers/d-icon";
+import { ajax } from "discourse/lib/ajax";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import { i18n } from "discourse-i18n";
+
+class DiscourseIdAdmin extends Component {
+  @service dialog;
+  @service toasts;
+
+  @tracked isRegenerating = false;
+  @tracked enabled;
+
+  constructor() {
+    super(...arguments);
+    this.enabled = this.args.model.enabled;
+  }
+
+  get hasCustomProvider() {
+    const url = this.args.model.provider_url;
+    return url && url !== "https://id.discourse.com";
+  }
```

Comment:
The custom provider setting is only for testing. We shouldn't expose it to end users IMO. (Even on the app side, we should make it harder to override... it probably shouldn't be a site setting.)